vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/mapConnector.js

```diff
@@ -62,34 +62,51 @@ import { getLayerForFeature } from "./util";
 
     mapElement.configuration.on("singleclick", (event) => {
       const coordinate = event.coordinate;
+      // Get the features at the clicked pixel position
+      // In case multiple features exist at that position, OpenLayers
+      // returns the features sorted in the order that they are displayed,
+      // with the front-most feature as the first result, and the
+      // back-most feature as the last result
+      const pixelCoordinate = event.pixel;
+      const featuresAtPixel =
+        mapElement.configuration.getFeaturesAtPixel(pixelCoordinate);
+      // Create tuples of features and the layer that they are in
+      const featuresAndLayers = featuresAtPixel.map((feature) => {
+        const layer = getLayerForFeature(
+          mapElement.configuration.getLayers().getArray(),
```

Comment:
Not sure how costly this is, but I believe this array could be stored outside of the loop.

Reply:
It just [accesses the internal array](https://github.com/openlayers/openlayers/blob/v6.12.0/src/ol/Collection.js#L167), the OL Collection is basically an observable wrapper around an array. The only benefit for extracting a variable here might be readability, but I don't think it improves much.

Reply:
That's fine, then.